level_5/lexer.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#include <iostream>
+#include <stdexcept>
+#include <variant>
+#include <cctype>
+
+enum class Type {
+	INTEGER, PLUS, MINUS, MUL, DIV, END, LPARAN, RPARAN
```

Comment:
```suggestion
	Integer, Add, Sub, Mul, Div, LeftParen, RightParen, Eof, Invalid
```

Using all-caps enum values dates back to pre-C++11 where there was no `enum class`. No need for doing that anymore.